packages/alea-frontend/pages/guided-tour2/[id].tsx

```diff
@@ -240,13 +252,39 @@ async function stateTransition(
   const newState = { ...state };
   const newMessages: ChatMessage[] = [];
   let nextAction: UserAction = undefined;
+  let nextConceptUri;
   const currentConceptUri = state.leafConceptUris[state.focusConceptIdx];
-  const nextConceptUri = state.leafConceptUris[state.focusConceptIdx + 1];
+  if (state.focusConceptIdx + 1 === state.leafConceptUris.length) {
+    const resp = await getLeafConcepts(state.targetConceptUri);
+    const lCUris = resp['leaf-concepts'] ?? [];
+    nextConceptUri = lCUris[0];
+  } else nextConceptUri = state.leafConceptUris[state.focusConceptIdx + 1];
   const currentConceptName = conceptUriToName(currentConceptUri);
   const nextConceptName = conceptUriToName(nextConceptUri);
 
   if (action.chosenOption === 'MOVE_ON') {
     newState.focusConceptIdx++;
+    newState.completedConceptUris.push(currentConceptUri);
+    const updatePayload: LearnerModelUpdateEvent = {
+      type: 'self-assessment',
+      concept: currentConceptUri,
+      competences: { Remember: 1.0, Understand: 1.0, Apply: 1.0 },
+      time: new Date().toISOString(),
+      payload: '',
+      comment: 'Self assessment through guided tour',
+    };
+    await updateLearnerModel(updatePayload);
+    if (
+      newState.completedConceptUris[newState.completedConceptUris.length - 1] ===
```

Comment:
Why not just check `if (currentConceptUri ===  newState.targetConceptUri)`

Reply:
implemented